src/Number.php

```diff
@@ -74,13 +74,13 @@ public function mul(string|int|float|BigNumber|Number $value): Number
         return static::of($this->value->multipliedBy($value->value));
     }
 
-    public function div(string|int|float|BigNumber|Number $value): Number
+    public function div(string|int|float|BigNumber|Number $value, int $decimalPlaces = 2): Number
```

Comment:
instead of defaulting to 2 decimal places here, what if we default it to `null` here and then the default decimal places is what is applied via the `of` method 👀

Reply:
Seen as the `div` method is the only method that requires knowledge of decimal places, I feel like placing it on the number is too heavy. For example, what happens if you use two different decimal places on the dividend and divisor?

Reply:
Actually the underlying `BigDecimal` object contains a scale already, so in the `of` method we could just call `->toScale($decimalPlaces)`

Reply:
@olivernybroe any thoughts on my previous statement?